scalafix-core/shared/src/main/scala/scalafix/internal/rule/OptionConstructor.scala

```diff
@@ -0,0 +1,28 @@
+package scalafix.internal.rule
+
+import scala.meta._
+import scala.meta.tokens.Token.LeftBrace
+
+import scalafix.Patch
+import scalafix.rule.{Rule, RuleCtx, RuleName}
+import scalafix.util.Trivia
+
+case object OptionConstructor
+    extends Rule(
+      RuleName("OptionConstructor").withDeprecatedName(
+        "OptionConstructor",
+        "Use the Option constructor instead",
+        "0.6.0"
+      )
+    ) {
+
+  override def description: String =
+    "Rewrite that replaces Some() constructor by Option() constructor"
+
+  override def fix(ctx: RuleCtx): Patch = {
+    ctx.tree.collect {
+      case b @ q"Some($a)" =>
```

Comment:
This will match `Some` syntactically, which is not safe since we're not sure it's `scala.Some`.

This has to be a `SemanticRule` and use `index.symbol(tree)` to query for the resolved symbol and check that is `scala.Some`.

A good test for this is 

```scala
import scala.{Some => MySome}

MySome(42) // the rule must be able to replace this with Option(42)
```